### ISSUE TITLE
aws-lambda-python-alpha: BundlingOptions exclude all assets and explicit include

### ISSUE THREAD

**Issue by MayconFrr** [aws/aws-cdk#30652]
### Describe the feature

Add a mode to BundlingOptions where it will import only a list of explicit written list of files/directories as opposed to the current pattern using asset_excludes

### Use Case

I propose adding a mode to the BundlingOptions in the CDK that allows for specifying assets to include explicitly, rather than excluding assets. This inclusion mode will simplify configurations where only a few files or directories need to be bundled, enhancing clarity and reducing configuration complexity. 

This feature will benefit AWS Lambda functions by potentially reducing deployment package size, thereby improving cold start times.

Here's some example of my current configuration and as can be seen, this can be tedious as more libraries are included

```python
PythonFunction(
            **other_atributes,
            bundling=BundlingOptions(
                asset_excludes=[
                    ".git",
                    ".idea",
                    ".mypy_cache",
                    ".pytest_cache",
                    ".ruff_cache",
                    ".venv",
                    ".vscode",
                    "cdk.out",
                    "cdk",
                    "node_modules",
                    "tests",
                    ".coverage",
                    ".coveragerc",
                    ".gitignore",
                    ".pre-commit-config.yaml",
                    "app.py",
                    "bitbucket-pipelines.yml",
                    "cdk.context.json",
                    "cdk.json",
                    "coverage.xml",
                    "package.json",
                    "package-lock.json",
                    "poetry.toml",
                    "README.md",
                ]
            ),
        )
```

### Proposed Solution

Add an asset_includes attribute to BundlingOptions that works similarly to asset_excludes but specifies which assets to include, and have both options mutually exclusive

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.144.0

### Environment details (OS name and version, etc.)

---

**Comment by pahud** [comment]
Makes sense to me. Please help us prioritize this feature request with 👍
